design/backup-repo-config.md

```diff
@@ -0,0 +1,105 @@
+Therefore, we create a BackupRepository configMap as a template of the configurations to be applied to the backup repository CR when it is created by the BackupRepository controller. For an existing BackupRepository CR, the configMap is never visited, if users want to modify the configuration value, they should directly edit the BackupRepository CR.   
+The BackupRepository configMap is created by users. If the configMap is not there, nothing is specified to the backup repository CR, so the Unified Repo modules use the hard-coded values to configure the backup repository.  
+The BackupRepository configMap is backup repository type specific, that is, for each type of backup repository, there is one configMap. Therefore, a ```backup-repository-config``` label should be applied to the configMap with the value of the repository's type. During the backup repository creation, the configMap is searched by the label against the repository type, if no configMap is found, nothing is set to the backup repository CR.  
+
+### Configurations
+
+With the above mechanisms, any kind of configuration could be added. Here list the configurations defined at present:  
+```cache-limit-mb```: specify the size limit(in MB) for the local data cache. The more data is cached locally, the less data may be downloaded from the backup storage, so the better performance may be achieved. Practically, users can specify any size that is smaller than the free space so that the disk space won't run out. This parameter is for each repository connection, that is, users could change it before connecting to the repository. If a backup repository doesn't use local cache, this parameter will be ignored. For Kopia repository, this parameter is supported.  
+```enable-compression```: specify to enable/disable compression for a backup repsotiory. Most of the backup repositories support the data compression feature, if it is not supported by a backup repository, this parameter is ignored. Most of the backup repositories support to dynamically enable/disable compression, so this parameter is defined to be used whenever creating a write connection to the backup repository, if the dynamically changing is not supported, this parameter will be hornored only when initializing the backup repository. For Kopia repository, this parameter is supported and can be dynamically modified.  
+
+Below is an example of the BackupRepository configMap with the configurations:  
+```json
+{
+    "repoConfig": {
+        "cache-limit-mb": 2048,
+        "enable-compression": true
+    }
+}
+```
+
+To create the configMap, users need to save something like the above sample to a json file and then run below commands:  
+```
+kubectl create cm <config-name> -n velero --from-file=<json file name>
+kubectl label cm <config-name> -n velero backup-repository-config=<repository type>
```

Comment:
I think it's also worth mentioning different way of creating ConfigMap:

File: config-map.yaml
```yaml

apiVersion: v1
kind: ConfigMap
metadata:
  name: <config-name>
  namespace: velero
  labels:
    backup-repository-config: <repository type>
data:
  repoConfig: |
    {
      "cache-limit-mb": 2048,
      "enable-compression": true
    }
```

```shell
$ kubectl apply -f config-map.yaml